### PR TITLE
Bug 1170236 - Get correct key management r=gasolin

### PR DESCRIPTION
**Pull request by evanxd**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=36104bf3741be578d0de9d20661d6c32c0618036

File: apps/settings/js/modules/wifi_utils.js [hunk 1]
```diff
@@ -195,12 +194,14 @@ define(function(require) {
...
         };
 
+        var key = WifiHelper.getKeyManagement(network);
         submitButton.disabled =
           !WifiHelper.isValidInput(key, password.value, identity.value,
             eap.value) || isSSIDInvalid();
       };
 
       eap.onchange = function() {
+        var key = WifiHelper.getKeyManagement(network);
```
Comment (evelynhung) [inline apps/settings/js/modules/wifi_utils.js]: Why we need to duplicate this line?

Comment (evelynhung) [inline apps/settings/js/modules/wifi_utils.js]: I guess my question is: would the `network` object will change within this function scope?

Comment (evanxd) [inline apps/settings/js/modules/wifi_utils.js]: Yes, `network` object will be changed if user changes "Security" type in "Authentication" page.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a37a7cb57f9fb4135edfa9f5e5e7640364812eb2

File: apps/settings/js/modules/wifi_utils.js [hunk 2]
```diff
@@ -187,20 +186,14 @@ define(function(require) {
...
-
+        var key = WifiHelper.getKeyManagement(network);
         submitButton.disabled =
           !WifiHelper.isValidInput(key, password.value, identity.value,
-            eap.value) || isSSIDInvalid();
+            eap.value) || !WifiHelper.isSSIDValid(ssid.value);
       };
 
       eap.onchange = function() {
+        var key = WifiHelper.getKeyManagement(network);
```
Comment (evanxd) [inline apps/settings/js/modules/wifi_utils.js]: The `network` object will be changed if user changes "Security" type in "Authentication" page.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5904153091bbde5a8f85c99a27ab61e12b1a1cc5

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c18a2cc29ffa3a9690c325dd5f65f5a653572684

File: shared/js/wifi_helper.js [hunk 3]
```diff
@@ -143,6 +143,14 @@ var WifiHelper = {
     return true;
   },
 
+  isSSIDValid: function(ssid) {
+    if (!ssid || ssid.length === 0) {
```
Comment (gasolin) [inline shared/js/wifi_helper.js]: how about `return (!ssid || ssid.length === 0)`

**Comment by gasolin** [comment]
and please git --amend comment to `r=gasolin`

File: shared/js/wifi_helper.js (hunk 3 above)
Comment (evanxd) [inline shared/js/wifi_helper.js]: Sure.

**Comment by evanxd** [comment]
Sure.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cccab77cc47bd4a7ce2df0e48dcb03ffb0c5a524

**Pull request merged**